channeldb/invoices.go

```diff
@@ -848,7 +838,23 @@ func (k *kvInvoiceUpdater) Finalize(updateType invpkg.UpdateType) error {
 		return k.storeSettleHodlInvoiceUpdate()
 
 	case invpkg.CancelInvoiceUpdate:
-		return k.serializeAndStoreInvoice()
+		err := k.serializeAndStoreInvoice()
```

Comment:
Couldn't we instead just call `k.storeCancelHtlcsUpdate()`? Which makes me think why we didn't do that in the first place? Maybe there's a reason? Missing too much context to be sure...

Reply:
good call, changed it makes absolutely sense to also persist the AMP state. We never changed the AMP related stuff when canceling an invoice, which is acutally the bug this PR is fixing.